lightningd/lightningd.c

```diff
@@ -251,6 +253,32 @@ static void daemonize_but_keep_dir(void)
 	tal_free(cwd);
 }
 
+static void pidfile_create(const struct lightningd *ld)
+{
+        int pid_fd;
+        char *pid;
+        const tal_t *tmpctx = tal_tmpctx(NULL);
+
+        /* Create PID file */
+        pid_fd = open(ld->pidfile, O_WRONLY|O_CREAT, 0640);
+        if (pid_fd < 0)
+                err(1, "Failed to open PID file: %s", strerror(errno));
+
+        /* Lock PID file */
+        if (lockf(pid_fd, F_TLOCK, 0) < 0) {
+                /* Problem locking file */
+                err(1, "lightningd already running? Error locking PID file: %s", strerror(errno));
+        } else {
```

Comment:
`err` will exit the program anyway, so no need for `else` branch here.  Just write the `pid = tal_fmt ...` etc directly after the closing bracket of the "then" branch.